src/test/java/uk/gov/ons/fwmt/job_service/rest/impl/FieldPeriodResourceServiceImplTest.java

```diff
@@ -54,10 +56,17 @@ public void findByFieldPeriodAndThrowHttpClientErrorException() {
     when(restTemplate.getForEntity(any(), eq(FieldPeriodDto.class), eq(fieldPeriod)))
         .thenThrow(new HttpClientErrorException(HttpStatus.BAD_REQUEST));
 
+    ResourceServiceMalfunctionException exception = null;
+
     //When
-    Optional<FieldPeriodDto> result = fieldPeriodResourceService.findByFieldPeriod(fieldPeriod);
+    try {
+      Optional<FieldPeriodDto> result = fieldPeriodResourceService.findByFieldPeriod(fieldPeriod);
+    } catch (ResourceServiceMalfunctionException e) {
```

Comment:
don't catch exception in test.

Use ExpectedException and @Rule to verify the code and exception. 
CSVParsingServiceImplTest has a test that does it.

Reply:
This has been implemented.